app/main.py

```diff
@@ -1,6 +1,16 @@
 from typing import Callable
 
+cache_dict = {}
```

Comment:
Create `cache_dict` as a class attribute and do not use `global` when calling it